hypervisor/lib/crypto/mbedtls/sha256.c

```diff
@@ -176,10 +173,11 @@ int32_t mbedtls_internal_sha256_process( mbedtls_sha256_context *ctx,
         P( A[3], A[4], A[5], A[6], A[7], A[0], A[1], A[2], W[i+5], K[i+5] );
         P( A[2], A[3], A[4], A[5], A[6], A[7], A[0], A[1], W[i+6], K[i+6] );
         P( A[1], A[2], A[3], A[4], A[5], A[6], A[7], A[0], W[i+7], K[i+7] );
+
+        i += 8;
```

Comment:
Moving the loop counter inside loop body violates another rule.
Modification of loop counter in loop body.
A loop counter used within a for loop for iteration shall not be modified in the body of the loop. Modification of the loop counter makes code difficult to read and maintain.

One suggestion:
add a local variable inside loop body to get the right index from the loop counter.

Reply:
what is about keep the original code? I don't think it is complex code.